File: backend/candidate-profile-service/src/main/java/com/jobportal/candidateprofile/controllers/CandidateEducationController.java
package com.jobportal.candidateprofile.controllers;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.jobportal.candidateprofile.dto.CandidateEducationDTO;
import com.jobportal.candidateprofile.entities.CandidateEducation;
import com.jobportal.candidateprofile.service.CandidateEducationService;

@RestController
@RequestMapping("/education")
@CrossOrigin("*")
public class CandidateEducationController {
	
	@Autowired
    private CandidateEducationService service;

    // Add Education
    @PostMapping("/add")
    public CandidateEducation addEducation(@RequestBody CandidateEducationDTO dto) {
        return service.addEducation(dto);
    }

    // Get All
    @GetMapping("/all")
    public List<CandidateEducation> getAllEducation() {
        return service.getAllEducation();
    }

    // Get By Candidate ID
    @GetMapping("/candidate/{cid}")
    public List<CandidateEducation> getEducationByCandidate(@PathVariable int cid) {
        return service.getEducationByCandidate(cid);
    }

    // Get By Id
    @GetMapping("/{id}")
    public CandidateEducation getEducationById(@PathVariable int id) {
        return service.getEducationById(id);
    }

    // Update
    @PutMapping("/update/{id}")
    public CandidateEducation updateEducation(
            @PathVariable int id,
            @RequestBody CandidateEducationDTO dto) {

        return service.updateEducation(id, dto);
    }

    // Delete
    @DeleteMapping("/delete/{id}")
    public String deleteEducation(@PathVariable int id) {
        return service.deleteEducation(id);
    }
}

File: backend-dotnet/Services/JobPortal.CandidateService/Controllers/CandidateEducationController.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using JobPortal.Shared.Entities;
using JobPortal.CandidateService.DTOs;
using JobPortal.CandidateService.Services;

namespace JobPortal.CandidateService.Controllers;

[ApiController]
[Route("education")]
public class CandidateEducationController : ControllerBase
{
    private readonly CandidateEducationService _service;

    public CandidateEducationController(CandidateEducationService service)
    {
        _service = service;
    }

    [HttpPost("add")]
    public async Task<IActionResult> AddEducation([FromBody] CandidateEducationDTO dto)
    {
        var result = await _service.AddEducationAsync(dto);
        return Ok(result);
    }

    [HttpGet("all")]
    public async Task<IActionResult> GetAllEducation()
    {
        var list = await _service.GetAllEducationAsync();
        return Ok(list);
    }

    [HttpGet("candidate/{cid}")]
    public async Task<IActionResult> GetEducationByCandidate(int cid)
    {
        var list = await _service.GetEducationByCandidateAsync(cid);
        return Ok(list);
    }

    [HttpGet("{id}")]
    public async Task<IActionResult> GetEducationById(int id)
    {
        var result = await _service.GetEducationByIdAsync(id);
        if (result == null) return NotFound("Education record not found");
        return Ok(result);
    }

    [HttpPut("update/{id}")]
    public async Task<IActionResult> UpdateEducation(int id, [FromBody] CandidateEducationDTO dto)
    {
        var result = await _service.UpdateEducationAsync(id, dto);
        if (result == null) return NotFound("Education record not found");
        return Ok(result);
    }

    [HttpDelete("delete/{id}")]
    public async Task<IActionResult> DeleteEducation(int id)
    {
        var msg = await _service.DeleteEducationAsync(id);
        if (msg == "Education Not Found") return NotFound(msg);
        return Ok(msg);
    }
}
